app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -31,41 +30,32 @@ func (s *accessLogServer) NeedLeaderElection() bool {
 func NewAccessLogServer(dataplane kumadp.Dataplane) *accessLogServer {
 	address := envoy.AccessLogSocketName(dataplane.Name, dataplane.Mesh)
 	return &accessLogServer{
-		server:  grpc.NewServer(),
 		address: address,
 	}
 }
 
 func (s *accessLogServer) Start(stop <-chan struct{}) error {
-	v3.RegisterAccessLogServer(s.server)
-
-	_, err := os.Stat(s.address)
-	if err == nil {
-		// File is accessible try to rename it to verify it is not open
-		newName := s.address + ".bak"
-		err := os.Rename(s.address, newName)
-		if err != nil {
-			return errors.Errorf("file %s exists and probably opened by another kuma-dp instance", s.address)
-		}
-		err = os.Remove(newName)
-		if err != nil {
-			return errors.Errorf("not able the delete the backup file %s", newName)
-		}
+	alStreamer := v3.NewAccessLogStreamer()
+	os.Remove(s.address)
+	err := syscall.Mkfifo(s.address, 0666)
+	if err != nil {
+		return errors.Wrapf(err, "error creating fifo %s", s.address)
 	}
-
-	lis, err := net.Listen("unix", s.address)
+	fd, err := os.OpenFile(s.address, os.O_CREATE, os.ModeNamedPipe)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "error opening fifo %s", s.address)
 	}
 
+	reader := bufio.NewReader(fd)
+
 	defer func() {
-		lis.Close()
+		fd.Close()
 	}()
 
-	logger.Info("starting Access Log Server", "address", fmt.Sprintf("unix://%s", s.address))
+	logger.Info("starting Access Log Server", "pipefile", fmt.Sprintf("%s", s.address))
```

Comment:
it's not really a server anymore, right? Can we try to find a better name?

also: why `fmt.Sprintf("%s", s.address)` and not just `s.address`